Split parameters at unquoted whitespaces

markusdy opened the pull request:
Hi there, 

currently, jvm options and jboss arguments from arquillian.xml are splited on whitespaces ("\\s+").
Even though I know that whitespaces in path names are evil, they do happend. This pull requests changes the way the parameter string from arquillian.xml is splited to the default way jvm arguments are read in java, that is to split on whitespaces if they are not quoted in quotes.
Example: foo bar "foo bar" will be split to [foo, bar, foo bar] instead of [foo, bar, "foo, bar"].

Attention:
I created org.jboss.as.arquillian.container.ParameterUtils to do the spliting. The class resides in wf-arq-common, since I first thought, that's the right place for such a util class (since there is also NetworkUtils). 
Later on, I wanted to change all the .split("\\s+") occurences, but I failed at ManagedDomainDeployabeContainer, since the wf-arq-container-domain-managed project has no common dependency with the other projects. 
As a result, I left MDDContainer unchanged and would like to get your feedback first, how to change the file without copy-paste ParameterUtils to wf-arq-common-domain (or maybe if it is the way you want it, I would also do that).

Greetings

jamezp commented on common/src/main/java/org/jboss/as/arquillian/container/ParameterUtils.java:
```diff
@@ -0,0 +1,54 @@
+     * @param parameterString
+     * @return
+     */
+    public static ArrayList<String> splitParams(String parameterString) {
```
This could just return a `List`, `Collection` or even just `Iterable`. We shouldn't return a concrete type like this.

jamezp commented on container-managed/src/main/java/org/jboss/as/arquillian/container/managed/ManagedDeployableContainer.java:
```diff
@@ -100,7 +101,7 @@ protected void startInternal() throws LifecycleException {
             commandBuilder.setJavaHome(config.getJavaHome());
             if (javaOpts != null && !javaOpts.trim().isEmpty()) {
-                commandBuilder.setJavaOptions(javaOpts.split("\\s+"));
+                commandBuilder.setJavaOptions(ParameterUtils.splitParams(javaOpts).toArray(new String[0]));
```
There's not need to export this to an array. There is a `setJavaOptions(Iterable<String>)` method.

jamezp commented on container-managed/src/main/java/org/jboss/as/arquillian/container/managed/ManagedDeployableContainer.java:
```diff
@@ -116,7 +117,7 @@ protected void startInternal() throws LifecycleException {
 
             if (jbossArguments != null && !jbossArguments.trim().isEmpty()) {
-                commandBuilder.addServerArguments(jbossArguments.split("\\s+"));
+                commandBuilder.addServerArguments(ParameterUtils.splitParams(jbossArguments).toArray(new String[0]));
```
There's not need to export this to an array. There is a `setJavaOptions(Iterable<String>)` method.

markusdy commented:
Ouch. Returning an ArrayList seriously hurts, I'm sorry! I fixed that and removed the use of the `toArray`.
I also duplicated the class for domain mode.

Concerning the wildfly-maven-plugin I took a brief look (as I use that one, too), the parameter `String jvmArgs` is deprecated and was changed to `String[] javaOpts` so at least with the latter one it should work. For some reason, it was changed only in `RunMojo`and `StartMojo` but not in `ExecuteCommandsMojo` though.

markusdy commented:
Sorry for the commit mess, in case of merging I suggest to squash commits.

jamezp commented:
Great thanks for the changes! I'll have another look here soon.

With regards to the wildfly-maven-plugin, thanks I completely forgot I switched that to an array rather than a single string :) I'll have a look at why it wasn't changed in both the `run` and `start` goals.
